### PR TITLE
Tests: Refactor test helper code.

### PR DESCRIPTION
**Pull request by jodh-intel**
Replace the two similar scripts...

        tests/integration/lib/common.bash
        tests/metrics/common/test.common

... with a single common one:

        tests/lib/test-common.bash

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

File: tests/integration/lib/common.bash [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-	fi
-}
-
-#Checking that default runtime is cor
-function runtimeDocker(){
-	default_runtime=`$DOCKER_EXE info 2>/dev/null | grep "^Default Runtime" | cut -d: -f2 | tr -d '[[:space:]]'`
-	if [ "$default_runtime" != "cor" ]; then
-		skip "Tests need to run with cor as default runtime"
-	fi
-}
```
Comment (jcvenegas) [inline tests/integration/lib/common.bash]: nice :)

**Comment by jcvenegas** [comment]
lgtm

**Comment by jodh-intel** [comment]
Hi @chavafg - looks like this PR hit the infamous #327 bug (now fixed by #486). Could you restart the tests to see if they are now happy with this PR?

**Comment by chavafg** [comment]
@jodh-intel, sure, restarting tests for this PR

**Comment by chavafg** [comment]
@jodh-intel , still failing, but it seems that it requires a rebase as I do not see #486 merged in jodh-intel:refactor-test-helper-code

**Comment by dlespiau** [comment]
We should probably rebase PRs, not merge master into them.

**Comment by jodh-intel** [comment]
@dlespiau - fixed :).

@chavafg - should work now :)

**Comment by jodh-intel** [comment]
`clang` under Travis had a bit of a fit, but restarting the job shows it's now happier.

 Looking at the `clang` log (`gcc` doesn't complain), I've just noticed a bunch of non-fatal warnings from `check.h`:

```
/home/travis/.cache/cor/include/check.h:454:75: warning: token pasting of ','
      and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
```

**Comment by gorozco1** [comment]
lgtm

**Comment by jodh-intel** [comment]
Hi @chavafg - could you re-spin the tests for this PR once more? Sorry for the trouble!

**Comment by chavafg** [comment]
@jodh-intel  sure, relaunching tests.

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**